test/extensions/secondorder/hbp/test_kfac.py

```diff
@@ -23,3 +31,48 @@ def test_kfac_not_supported(problem):
         BackpackExtensions(problem).kfac()
 
     problem.tear_down()
+
+
+@pytest.mark.montecarlo
+@pytest.mark.parametrize("problem", BATCH_SIZE_1_PROBLEMS, ids=BATCH_SIZE_1_IDS)
+def test_kfac_should_approx_ggn_montecarlo(problem):
+    """Check that for batch_size = 1, the K-FAC is the same as the GGN.
+
+    Args:
+        problem (ExtensionsTestProblem): Test case.
+    """
+    problem.set_up()
+    # calculate GGN
+    autograd_res = AutogradExtensions(problem).ggn_blocks()
+
+    # calculate backpack average
+    mc_samples = 300000
+    backpack_kfac = BackpackExtensions(problem).kfac(mc_samples)
+    backpack_res = [kfacs_to_mat(kfac) for kfac in backpack_kfac]
+
+    # check the values
+    check_sizes_and_values(autograd_res, backpack_res, atol=1e-1, rtol=1e-1)
+
+    problem.tear_down()
+
+
+@pytest.mark.parametrize("problem", BATCH_SIZE_1_PROBLEMS, ids=BATCH_SIZE_1_IDS)
+def test_kfac_should_approx_ggn_montecarlo_light(problem):
+    """Check that for batch_size = 1, the K-FAC is the same as the GGN.
+
+    Args:
+        problem (ExtensionsTestProblem): Test case.
+    """
+    problem.set_up()
+    # calculate GGN
+    autograd_res = AutogradExtensions(problem).ggn_blocks()
+
+    # calculate backpack average
+    mc_samples = 6000
+    backpack_kfac = BackpackExtensions(problem).kfac(mc_samples)
+    backpack_res = [kfacs_to_mat(kfac) for kfac in backpack_kfac]
+
+    # check the values
+    check_sizes_and_values(autograd_res, backpack_res, atol=1e-1, rtol=1e-1)
```

Comment:
Maybe reduce to `1e-2` if it works on all problems.

Reply:
TODO: Verify it works on GPU

Reply:
Reduced tolerances in commit
"Fixed shape of ggn blocks, type annotation, tol sizes, and commenting"

Reply:
(See comment on non-light version of this problem for full description of the updated info)

Technically this version passed on the GPU--but barely. More comfortable passing threshold would be 1e-1 tolerance for 6e3 samples, or 6e4 samples for 1e-2 tolerance

Reply:
Let's stick to the current tolerances then. Thanks for the report. Marking as resolved.